src/Featurizers/PCAFeaturizer.h

```diff
@@ -0,0 +1,267 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include "../Featurizer.h"
+#include "../Archive.h"
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         PCATransformer
+///  \brief         Contains PCAComponents and use PCAComponents to project
+///                 matrix for dimensionality reduction, also provides
+///                 PCAComponents retriving
+///
+template <typename MatrixT>
+class PCATransformer : public StandardTransformer<MatrixT, MatrixT> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType                                   = StandardTransformer<MatrixT, MatrixT>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    PCATransformer(MatrixT eigenvalues, MatrixT eigenvectors);
+    PCATransformer(Archive &ar);
+
+    ~PCATransformer(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(PCATransformer);
+
+    bool operator==(PCATransformer const &other) const;
+
+    void save(Archive &ar) const override;
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    MatrixT const                                    _eigenvalues;
```

Comment:
>MatrixT const                                    _eigenvalues; [](start = 4, length = 62)

Is this being used?

Reply:
Not for transformer but for potential annotation(eigenvalues) retrieving

Reply:
If that is the case, create an Estimator in Components and have this Estimator use that one. Because of how close we are to release, I'd recommend removing the value rather than going through a refactor like this.

Reply:
ok